src/geometry/vector2d.ts

```diff
@@ -0,0 +1,157 @@
+import { Integer, int } from "utils";
+
+export enum Symmetry {
+  ORIGIN = 0,
+  X_AXIS = 1,
+  Y_AXIS = 2
+}
+
+
+export class Vector2d {
+  protected readonly x: number;
+  protected readonly y: number;
+
+  protected createVector(x: number, y: number): Vector2d {
+    return new Vector2d(x, y);
+  }
+
+  public constructor(x: number, y: number) {
+    this.x = x;
+    this.y = y;
+  }
+
+  public getX() {
+    return this.x;
+  }
+
+  public getY() {
+    return this.y;
+  }
```

Comment:
For new we don't need getters, also TypeScript offers explicit getters and setters: [TypeScript getters/setters](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters).